### PR TITLE
Enable Service Affinity for OpenStack cloudprovider

### PR DESCRIPTION
**Pull request by FengyunPan**
Fix issue: #48500
Kubernetes's OpenStack cloudprovider can't set persistence to "SOURCE_IP"

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go [hunk 1]
```diff
@@ -546,15 +546,6 @@ func (lbaas *LbaasV2) createLoadBalancer(service *v1.Service, name string) (*loa
 	return loadbalancer, nil
 }
 
-func stringInArray(x string, list []string) bool {
```
Comment (dims) [inline pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go]: This seems like an unrelated change. Can you please file as a separate PR?

Comment (FengyunPan) [inline pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go]: Ok, done.

**Comment by anguslees** [comment]
Oops ;)
/lgtm
